app/controllers/programs/managers_controller.rb

```diff
@@ -0,0 +1,65 @@
+class Programs::ManagersController < ApplicationController
+  before_action :set_manager_program
+  before_action :set_manager, only: %i[ new edit_program_managers ]
+  include ManagerApi
+
+  def new
+  end
+
+  def edit_program_managers
+  end
+
+  def create
+    if params[:manager_id].present?
+      @manager = Manager.find(params[:manager_id])
+      authorize @manager
+      if @manager_program.managers << @manager
+        @manager = Manager.new
+        flash[:notice] = "The manager was added."
+      end
+    else
+      uniqname = manager_params[:uniqname]
+      result = get_name(uniqname, @manager_program)
```

Comment:
You could call LdapLookup directly here and is the return is false block the creation?

Reply:
The get_name() checks not only LdapLookup for uniqname validation. It also checks if the uniqname is not an instructor for the program in question. It returns valid = false in both cases with a corresponding alert message. And in case of empty last and first names, it returns a note message.